### PR TITLE
Update motus to 2.0.0 (#9225)

### PR DESCRIPTION
**Pull request by unode**
* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

This recipe currently targets https://github.com/unode/mOTUs_v2 my personal fork of the actual tool that lives in https://github.com/motu-tool/mOTUs_v2.
An official 2.0.0 release is to be made available soon at which point I will update the recipe to point to the corresponding release and drop WIP from the pull request.

The tool includes a data download of ~1GB setup through `post-link.sh` as recommended in #9225 .
It wasn't clear if this data would be automatically removed so a `pre-unlink.sh` is included that removes it.
Since this is my first contribution, feedback is most welcome.

### CODE REVIEW

File: recipes/motus2/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  name: motus2
+  version: {{ version }}
+
+source:
+  url: https://github.com/unode/mOTUs_v2/archive/{{ version }}.tar.gz
+  sha256: bad994a73ab638b1b91fdfd165a82b608f32962015b1e8e89660b837aed34932
+
+build:
+  number: 0
+  skip: True # [not linux64]
```
Comment (bgruening) [inline recipes/motus2/meta.yaml]: [osx] would be better I think.

Comment (unode) [reply recipes/motus2/meta.yaml]: Will do

**Comment by unode** [comment]
@bgruening For testing purposes I also called this tool motus2.
In fact we'd like to submit it as an update to the already existing [motus 1.0](https://bioconda.github.io/recipes/motus/README.html). (edited checkbox in description)

Given the first version was packaged by someone else (@pirovc @ypriverol @ArneKr) should I simply rename the files to overwrite https://github.com/bioconda/bioconda-recipes/tree/master/recipes/motus ?

**Comment by bgruening** [comment]
Can you remove the old recipe into a subdirctory called 1.0 and put yours into the main dir?

**Comment by unode** [comment]
Linting failures on the older motus 1.0.
From #3418 `perl-threaded` is now `perl`. Fixed here.

**Comment by unode** [comment]
This is now final.
Anything else needed from my side?

**Comment by unode** [comment]
Rebased to resolve conflicts introduced by 31f2bc3

**Comment by unode** [comment]
I don't know what `should_not_use_fn` means nor what to do to fix it.
In any case, the errors are on motus 1.0. Any chance to get motus 2.0 merged in the meanwhile?

**Comment by druvus** [comment]
@unode I fixed the fn issue in your recipe. It was due to changes in infrastructure.

**Comment by unode** [comment]
@druvus Thanks!

**Comment by AlessioMilanese** [comment]
Dear @bgruening, @druvus 

@unode initially opened the merge request for motus **v2.0.0**. Would it be possible to merge this version (which is in python) and then solve the problems (related to perl) with version 1.0.

**Comment by bgruening** [comment]
Lets see how it behaves with our new build-infrastructure.

**Comment by AlessioMilanese** [comment]
Dear @bgruening,

I see in
https://bioconda.github.io/recipes/motus/README.html

that there is still motus v1.0.
Is there something that we can do from our side?

**Pull request merged**